### PR TITLE
Add Logger

### PR DESCRIPTION
**Pull request by bh2smith**
This PR is split into two commits:

1. Another version bump of near-lake-frame with improvements related to logging. Specifically [this PR](https://github.com/frolvanya/near-lake-framework-py/pull/13) which also migrated the `fetch_latest_block` function from here to there.
2. Adding a logger and replacing all print statements with log statements. Note that we explicitly filter out the noisy "missing shard" log (this is only temporary because I think the missing shard log should be debug instead of warning).

### CODE REVIEW

File: logger.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        """Enable or disable the filter."""
+        self.enabled = enable
+
+    def get_ignored_message(self) -> str:
+        """Get the current ignored message pattern."""
+        return self.ignored_message
+
+
+def set_logger(name: str) -> logging.Logger:
+    logging.basicConfig(level=logging.INFO)
```
Comment (anthony-near) [inline logger.py]: IMO these fields that we're setting should be from config (or if you think that's overkill, then set via CONSTANTS), not set directly in this function. Should load from config or constants and set in this fn.

File: logger.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    logging.getLogger("near_lake_framework").setLevel(logging.INFO)
+    missing_shard_filter = IgnoreFilter("doesn't exist")
```
Comment (anthony-near) [inline logger.py:35]: Is the missing shard data something that gets backfilled in future data from near lake or skipped?

**Review by anthony-near — CHANGES_REQUESTED** [review]

File: logger.py (hunk 2 above)
Comment (bh2smith) [reply logger.py:35]: My understanding is that there was no data on the shared corresponding to the block. I think this filter can be removed if we downgrade this log to debug and set the lake log level to info. The filter is only a temporary solution.

Comment (bh2smith) [reply logger.py:35]: Created a PR here: https://github.com/frolvanya/near-lake-framework-py/pull/16

We can leave this open and wait for the next version bump (if its approved) - then remove the whole filter class.

File: logger.py (hunk 1 above)
Comment (bh2smith) [reply logger.py]: I've made a minor adjustment to the log level configuration... 
https://github.com/near/gas-station-event-indexer/pull/10/commits/d77b745727a4901010b4dc0c27ef6e49fe4ed7e6

However doing this properly is going to be a much bigger change. So much that I would insist it be saved for follow up PR.

**Review by anthony-near — APPROVED** [review]
Once the linter is happy https://github.com/near/gas-station-event-indexer/actions/runs/9305341947/job/25612033263, then we can merge. Seems like the best long term solution is improvements on the near-lake-framework-py repo.

**Comment by bh2smith** [comment]
Linter is happy, I made an issue (https://github.com/near/gas-station-event-indexer/issues/11) related to removal of log filter and hopefully be back soon to do the log configuration more properly (https://github.com/near/gas-station-event-indexer/issues/12).

**Pull request merged**